### PR TITLE
Has one relationship

### PR DESCRIPTION
**Pull request by zaidakram**
My first ever contribution to this wonderful project. And thanks to all the contributors for the awesome stuff that is Amber.
 
This PR adds `has_one` relationship to the models. Adds two simple methods which are documented and tested.

**NOTE:** It does not support through associations yet.

### CODE REVIEW

File: docs/relationships.md [hunk 1]
```diff
@@ -60,6 +139,7 @@ CREATE TABLE posts (
 CREATE INDEX 'user_id_idx' ON posts (user_id);
 ```
 
+
```
Comment (Blacksmoke16) [inline docs/relationships.md]: Can we remove this extra line?

Comment (zaidakram) [reply docs/relationships.md]: Done!

File: src/granite/associations.cr [hunk 2]
```diff
@@ -36,6 +36,24 @@ module Granite::Associations
...
+    # retrieve the associated object
+    def {{model_name.id}}
+      {{model_name.id.camelcase}}.find_by({{foreign_key.id}}: self.id)
+    end
+
+    # set the associated object to given relationship
+    def {{model_name.id}}=(children)
+      children.{{foreign_key.id}} = self.id
+    end
+  end
```
Comment (robacarp) [inline src/granite/associations.cr]: The comments here are probably redundant. The code is pretty straightforward.

Comment (zaidakram) [reply src/granite/associations.cr]: Just removed them. You are right. Code is pretty self-explanatory.

**Pull request merged**